lib/util/resolve-base-url.js

```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+var url = require('url')
+
+module.exports = resolveBaseUrl
+
+function resolveBaseUrl(repository) {
+  var parsed = url.parse(repository)
+
+  if (parsed.host) {
+    return 'https://' + parsed.host + '/'
```

Comment:
I can't use `parsed.protocol` as it resolves to original protocol (e.g. `git://`), which is not the case.

Reply:
I'm afraid it has to be hardcoded in this fashion.

> The WHATWG URL Standard considers a handful of URL protocol schemes to be special in terms of how they are parsed and serialized. When a URL is parsed using one of these special protocols, the `url.protocol` property may be changed to another special protocol but cannot be changed to a non-special protocol, and vice versa.
> 
> Likewise, changing from a non-special protocol to a special protocol is also not permitted:
> ```js
> const u = new URL('fish://example.org');
> u.protocol = 'http';
> console.log(u.href);
> // fish://example.org
> ```

[(source)](https://nodejs.org/dist/latest-v12.x/docs/api/url.html#url_special_schemes)

Reply:
lol. That’s...

Can we work around it by doing something like: `new URL('http:' + url.href.slice(url.href.indexOf(':')))`?

I love url, but it’s a bit of a downside that it’s deprecated and the suggestion is to use URL

Reply:
Switched it back to concatenation. Alternatively I can use:
```js
 return new URL('/' + repo.user + '/' + repo.project, base) + '/'
```
which would be more foolproof, in case someone provide baseUrl with trailing slash `/`, leading to `https://github.com//such/urls`